replicate/common/generic-replicator.go

```diff
@@ -447,6 +511,13 @@ func (r *GenericReplicator) DeleteResources(source interface{}, list *v1.Namespa
 	}
 }
 
+// DeleteResourcesByLabels deletes resources in a list of namespaces acquired by evaluating namespace labels
+func (r *GenericReplicator) DeleteResourcesByLabels(source interface{}, list *v1.NamespaceList) {
```

Comment:
Naming: Does this function really "delete resources by labels"? From the looks of it, it "just" deletes a single resource (identified by name) in multiple namespaces.